src/devices/Tm1637/Tm1637.cs

```diff
@@ -0,0 +1,299 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device;
+using System.Device.Gpio;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.WindowsRuntime;
+
+namespace Iot.Device.Tm1637
+{
+    public class Tm1637 : IDisposable
+    {
+        /// <summary>
+        /// The number of segments that the TM1637 can handle
+        /// </summary>
+        public byte MaxSegments => 6;
+        // According to the doc, the clock pulse width minimum is 400 ns
+        // And waiting time between clk up and down is 1 µs
+        private const byte ClockWidthMicroseconds = 1;
+
+        private readonly int _pinClk;
+        private readonly int _pinDio;
+        private GpioController _controller;
+        private byte _brightness;
+        // Default segment order is from 0 to 5
+        private byte[] _segmentOrder = new byte[6] { 0, 1, 2, 3, 4, 5 };
+        // To store what has been displayed last. Used when change on brightness or
+        // screen on/off is used
+        private byte[] _lastDisplay = new byte[6] { 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
+        private bool _screenOn;
+
+        /// <summary>
+        /// Initialize a TM1637
+        /// </summary>
+        /// <param name="pinClk">The clock pin</param>
+        /// <param name="pinDio">The data pin</param>
+        /// <param name="pinNumberingScheme">Use the logical or physical pin layout</param>
+        public Tm1637(int pinClk, int pinDio, PinNumberingScheme pinNumberingScheme = PinNumberingScheme.Logical)
+        {
+            _pinClk = pinClk;
+            _pinDio = pinDio;
+            _controller = new GpioController(pinNumberingScheme);
+            _controller.OpenPin(_pinClk, PinMode.Output);
+            _controller.OpenPin(_pinDio, PinMode.Output);
+            _brightness = 7;
+        }
+
+        /// <summary>
+        /// Order of segments, expect a 6 length byte array
+        /// 0 to 5, any order. Most of the time 4 segments do
+        /// not need to be changed but the 6 ones may be in different order
+        /// like 0 1 2 5 4 3. In this case, this byte array has be be in this order
+        /// </summary>
+        public byte[] SegmentOrder
+        {
+            get { return _segmentOrder; }
+            set
+            {
+                if (value.Length != MaxSegments)
```

Comment:
does it have to be exactly 6? Shouldn't I be allowed to pass a byte[] of size 3 for example if I only really want to print 3 digits?

Reply:
Yes, for this one. Because you can have hardware where only the segments 3 to 5 are connected. So if you want them to be visible, you'll have to setup the full table anyway. 
What I've seen buying couple of those ones is that the ones with 4 segments used for clocks are usually all ok, 0 to 3 are usualy corresponding. So in those cases, you just Don't have to use this at all.
For the 6 segments ones, it's either all ok either order is 2 1 0 5 4 3.
And finally when you have only 3 segements, they are mostly ok but I've seen as well 5 4 3 as the first order. 
Finally, if in the case you have 3 segments, and in the case they are inversted in some ways, you'll pass Something like 5 4 3 0 1 2 where actually you Don't care or the real order of the last 3 ones.